nats-server/configuration/gateways/README.md

```diff
@@ -2,26 +2,30 @@
 
 ## Gateways
 
-Gateways enable connecting one or more clusters together; they allow the formation of super clusters from smaller clusters. Cluster and Gateway protocols listen in different ports. Clustering is used for adjacent servers; gateways are for joining clusters together. Typically all cluster nodes will also be gateway nodes, but this is not a requirement.
+Gateways enable connecting one or more clusters together into a full mesh; they allow the formation of super clusters from smaller clusters. Cluster and Gateway protocols listen on different ports. Clustering is used for adjacent servers; gateways are for joining clusters together. 
 
 Gateway configuration is similar to clustering:
 
 * gateways have a dedicated port where they listen for gateway requests
-* gateways gossip gateway members and remote discovered gateways
+* gateways gossip gateway nodes and remote discovered gateways
 
 Unlike clusters, gateways:
 
-* don't form a full mesh
+* have names, specifying the cluster they are in
+* don't form a full mesh between gateway nodes, but form a full mesh between cluster instead
 * are bound by uni-directional connections
+* don't gossip gateway nodes to clients
```

Comment:
I think I know what you mean here, but to be clear, nodes from a cluster are not gossiped to clients connected to another cluster. That is, if cluster A has 3 servers A1, A2 and A3, and cluster B has 2 servers B1 and B2, a client connecting to A1 will be notified of A2 and A3, but no server from B. Same for cluster B, a client connecting to B1 will be notified of B2 but no server from cluster A.

Reply:
This in the section where gateways are contrasted with cluster
Unlike clusters, gateways:
...
* don't gossip gateway nodes to clients

That is essentially what you are saying.